mw-webapp/src/hooks/useWayPreview.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from "react";
+import {useNavigate} from "react-router-dom";
+import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
+import {WayPreview} from "src/model/businessModelPreview/WayPreview";
+import {pages} from "src/router/pages";
+
+/**
+ * Fetches WayPreview and returns state
+ */
+export function useWayPreview (uuid: string) {
```

Comment:
Looks like this hook can do almost everything: load way, handling state, navigate, validate. Is it save to have this one? I'm not sure, but we should be careful here

Reply:
I'm not sure how to split this logic other way